sklearn/metrics/_scorer.py

```diff
@@ -65,19 +66,23 @@
 
 from ..utils.multiclass import type_of_target
 from ..base import is_regressor
+from ..utils._response import _get_response_values
 from ..utils._param_validation import validate_params
 
 
-def _cached_call(cache, estimator, method, *args, **kwargs):
+def _cached_call(cache, estimator, response_method, *args, **kwargs):
     """Call estimator with method and args and kwargs."""
     if cache is None:
-        return getattr(estimator, method)(*args, **kwargs)
-
+        return _get_response_values(
+            estimator, *args, response_method=response_method, **kwargs
+        )
     try:
-        return cache[method]
+        return cache[response_method]
     except KeyError:
-        result = getattr(estimator, method)(*args, **kwargs)
-        cache[method] = result
+        result = _get_response_values(
+            estimator, *args, response_method=response_method, **kwargs
+        )
```

Comment:
Looks like you are never using the pos_label returned by ``_get_response_values``. So now this function returns a tuple but only uses the first element. We should do ``result, _ = _get_response_values(...)``